src/PlayingBoard.re

```diff
@@ -0,0 +1,155 @@
+[%bs.raw {|require('./playingBoard.css')|}];
+open SharedTypes;
+
+type playingBoard = array(array(env));
+
+let addRock = (num: int, field: env) =>
+  switch (num) {
+  | 1 => Rock
+  | _ => field
+  };
+
+let addSpice = (num: int, field: env) =>
+  switch (field) {
+  | Sand =>
+    switch (num) {
+    | 1 => Spice
+    | _ => field
+    }
+  | _ => field
+  };
+
+let addHill = (num: int, field: env) =>
+  switch (field) {
+  | Rock =>
+    switch (num) {
+    | 1 => Hill
+    | _ => field
+    }
+  | _ => field
+  };
+
+let createSandBoard = ((width: int, height: int)) =>
+  Array.make_matrix(width, height, Sand);
+
+let randomizeBoard = (fn: (int, env) => env, seed: int, board: playingBoard) =>
+  Array.map(
+    row => Array.map(field => fn(Random.int(seed), field), row),
+    board,
+  );
+
+let getFieldType = ((y: int, x: int), board: playingBoard) =>
+  switch (y, x) {
+  | _ when y < 0 || y >= Array.length(board) => Empty
+  | _ when x < 0 || x >= Array.length(board[0]) => Empty
+  | _ => board[y][x]
+  };
+
+let compute = (env: env, sum: int, field: env) =>
+  env === field ? sum + 1 : sum;
+
+let countNeighbours =
+    (
+      (x, y),
+      getFieldFn: ((int, int), playingBoard) => env,
+      board,
+      countFn: (int, env) => int,
+    ) =>
+  List.fold_left(
+    (sum, (diffX, diffY)) => {
+      let neighbourY = y + diffY;
+      let neighbourX = x + diffX;
+      let field = getFieldFn((neighbourX, neighbourY), board);
+      countFn(sum, field);
+    },
+    0,
+    [
```

Comment:
This should be a constant. It's not necessary to pollute code with list of neighbours coordinates.